src/pfs/pfs.proto

```diff
@@ -284,6 +294,31 @@ message DeleteBranchRequest {
   bool force = 2;
 }
 
+message CreateProjectRequest {
+  Project project = 1;
+  string description = 2;
+  bool update = 3;
+}
+
+message CreateProjectResponse {}
+
+message InspectProjectRequest {
+  Project project = 1;
+}
+
+message InspectProjectResponse {
+  ProjectInfo project_info = 1;
+}
+
+message ListProjectRequest {}
+
+message DeleteProjectRequest {
+  Project project = 1;
+  bool force = 2;
```

Comment:
If `force` is true, should deleting a project which does not exist succeed or fail?

Reply:
I think you should have to pass `force` to delete a project's internal objects. Not passing force would only delete empty projects. WDYT?

Reply:
That makes sense to me — I assume by internal objects you mean repos & pipelines, right?

How much of this behaviour should be connected up in this PR vice a follow-up?

Reply:
given that repos and pipelines don't know where they belong yet, that should be done in a follow up. My intention here was to describe the API surface